charts/azuremonitor-containers/values.yaml

```diff
@@ -20,6 +20,20 @@ omsagent:
     pullPolicy: IfNotPresent
     dockerProviderVersion: "12.0.0-0"
     agentVersion: "1.10.0.1"
+
+  # The priority used by the omsagent priority class for the daemonset pods
+  # Note that this is not execution piority - it is scheduling priority, as
+  # in getting scheduled to the node.  This needs to be greater than 0 such
+  # that the daemonset pods, which can not schedule onto different nodes as
+  # they are defined to run on specific nodes, are not accidentally frozen
+  # out of a node due to other pods showing up earlier in scheduling.
+  # (DaemonSet pods by definition only are created once the node exists for
+  # them to be created for and thus it is possible to have "normal" pods
+  # already in line to run on the node before the DeamonSet controller got a
+  # chance to build pod for the node and give it to the scheduler)
+  # Should be some number greater than default (0)
+  priority: 10
```

Comment:
Why is default 10 ? Can we have 0 (no priority) if not specified thru params ?

Reply:
The reason for something larger than 0 is that at 0 this will end up having pods for some nodes stay pending due to the race I discussed if the cluster is dynamically scaling due to pending work pods.  If k8s had thought of this from the start, daemonset controller would have defaulted to something like 10 to begin with such that daemonsets are not frozen out.